contracts/src/v0.8/functions/dev/1_0_0/Functions.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {CBOR, Buffer} from "../../../shared/vendor/solidity-cborutils/v2.0.0/CBOR.sol";
+
+/**
+ * @title Library for Chainlink Functions
+ */
+library Functions {
+  uint16 public constant REQUEST_DATA_VERSION = 1;
+  uint256 internal constant DEFAULT_BUFFER_SIZE = 256;
+
+  using CBOR for Buffer.buffer;
+
+  enum Location {
+    Inline,
+    Remote,
+    DONHosted
+  }
+
+  enum CodeLanguage {
+    JavaScript
+    // In future version we may add other languages
+  }
+
+  struct Request {
+    Location codeLocation;
+    Location secretsLocation; // Only Remote secrets are supported
+    CodeLanguage language;
+    string source; // Source code for Location.Inline, url for Location.Remote or slot decimal number for Location.DONHosted
```

Comment:
This must be my bad, but we shall repeat the logic for `source` regarding `DONHosted` variant.
Maybe we can even add the same helper `addDONHostedSource()`